example/CMakeLists.txt

```diff
@@ -29,16 +27,24 @@ set(CUSTOM_PROXY_SOURCES
 
 # Add libmexclass as an external project.
 include(ExternalProject)
+# GIT_TAG for libmexclass source from Git.
+# Note: Clients who don't want to depend on main (which may have breaking changes) can specify a specific commit as argument with cmake command, else source from main branch will be used.
+if(DEFINED GIT_TAG)
```

Comment:
I would recommend renaming `GIT_TAG` to something like `LIBMEXCLASS_GIT_TAG` since `GIT_TAG` seems a bit generic.

Reply:
This is a minor suggestion, but I think this could be written a bit more concisely using a CMake cache entry and specifying the default value to `main`.

Reference: https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry